internal/ignition/ignition_test.go

```diff
@@ -813,69 +816,154 @@ var _ = Describe("downloadManifest", func() {
+			Expect(err).ToNot(HaveOccurred())
+
+			config, err := ParseToLatest([]byte(merge))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(config.Ignition.Version).To(Equal("3.2.0"))
+
+			bytes, err := json.Marshal(config)
+			Expect(err).ToNot(HaveOccurred())
+			v32Config, _, err := config_32.Parse(bytes)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(v32Config.Ignition.Version).To(Equal("3.2.0"))
+		})
+
+		It("does not parse v99 config", func() {
+			_, err := ParseToLatest([]byte(v99ignition))
+			Expect(err.Error()).To(ContainSubstring("unsupported config version"))
+		})
 
-		bytes, err := json.Marshal(config)
-		Expect(err).ToNot(HaveOccurred())
-		v32Config, _, err := config_32.Parse(bytes)
-		Expect(err).ToNot(HaveOccurred())
-		Expect(v32Config.Ignition.Version).To(Equal("3.2.0"))
 	})
 
-	It("does not parse v99 config", func() {
-		_, err := ParseToLatest([]byte(v99ignition))
-		Expect(err.Error()).To(ContainSubstring("unsupported config version"))
+	Context("using MergeStructTranscribe", func() {
+		It("v31 base with v31 override as 3.1.0", func() {
+			baseConfig, err := ParseToLatest([]byte(v31ignition))
+			Expect(err).ToNot(HaveOccurred())
+			overrideConfig, err := ParseToLatest([]byte(v31override))
+			Expect(err).ToNot(HaveOccurred())
+
+			mergeResult, _ := merge.MergeStructTranscribe(*baseConfig, *overrideConfig)
+			res, err := json.Marshal(mergeResult)
+			Expect(err).ToNot(HaveOccurred())
+
+			version, err := jsonparser.GetString(res, "ignition", "version")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(version).To(Equal("3.1.0"))
+		})
+
+		It("v31 base with v32 override as 3.2.0", func() {
+			baseConfig, err := ParseToLatest([]byte(v31ignition))
+			Expect(err).ToNot(HaveOccurred())
+			overrideConfig, err := ParseToLatest([]byte(v32override))
+			Expect(err).ToNot(HaveOccurred())
+
+			mergeResult, _ := merge.MergeStructTranscribe(*baseConfig, *overrideConfig)
+			res, err := json.Marshal(mergeResult)
+			Expect(err).ToNot(HaveOccurred())
+
+			version, err := jsonparser.GetString(res, "ignition", "version")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(version).To(Equal("3.2.0"))
+		})
+
+		It("v32 base with v32 override as 3.2.0", func() {
+			baseConfig, err := ParseToLatest([]byte(v32ignition))
+			Expect(err).ToNot(HaveOccurred())
+			overrideConfig, err := ParseToLatest([]byte(v32override))
+			Expect(err).ToNot(HaveOccurred())
+
+			mergeResult, _ := merge.MergeStructTranscribe(*baseConfig, *overrideConfig)
+			res, err := json.Marshal(mergeResult)
+			Expect(err).ToNot(HaveOccurred())
+
+			version, err := jsonparser.GetString(res, "ignition", "version")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(version).To(Equal("3.2.0"))
+		})
+
+		// Be aware, this combination is counterintuitive and comes from the fact that MergeStructTranscribe()
+		// is not order-agnostic and prefers the field coming from the override rather from the base.
```

Comment:
>prefers the field coming from the override rather from the base

This is actually something I'm a bit concerned about. Specifically, will switching to order to achieve the higher version end up causing issues when we want to actually override an existing value in the base?

Say we provide some file at `/tmp/example` in the base ignition at version 3.2 and someone provides an override for `/tmp/example` with version 3.1. What happens and what do we expect to happen?

I think the main thing we want to ensure is that the override file content is what ends up in the resulting ignition. Can you add a test for this case?

Reply:
I see the scenario... It's just by looking at our base template I'm very close to saying that if someone tries to override a value from the base template, they are doing something that's wrong...

I agree having `/tmp/example` in the base and in the override is problematic, but here we have more components

- our base hardcoded in the service
- base coming from the MCS (that we link as part of `merge` stanza)
- override provided by the user

If any 2 out of 3 above are trying to touch the same field, it's already a shady area